### PR TITLE
Basic character controller

### PR DESCRIPTION
**Pull request by pcanogo**
New important assets:
InputMaster (Input Actions Map) - Note: Don't look at the script, it's self generated
PlayerMain(Script)
PlayerCamera(Script)

### CODE REVIEW

File: Project Dinner/Assets/PlayerMain.cs [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    public CharacterController controller;
+    private InputMaster _controls;
+    public Transform groundCheck;
+    public LayerMask goundMask;
+    
+
+    public float speed = 12f;
+    public float gravity = -9.81f;
+    public float jumpHeight = 3f;
+    public float grounDistance = 0.4f;
```
Comment (dameos) [inline Project Dinner/Assets/PlayerMain.cs]: Change to `groundDistance`

File: Project Dinner/Assets/PlayerMain.cs [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+    private Vector2 _moveAxis;
+    private Vector3 _movement;
```
Comment (dameos) [inline Project Dinner/Assets/PlayerMain.cs:20]: Make this a local variable, you're only using in the `Update` method

File: Project Dinner/Assets/PlayerMain.cs [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+        /*Will merge these into 1 vector3 in future pr.*/
+        _movement = transform.right * _moveAxis.x + transform.forward * _moveAxis.y;
+        _velocity.y += gravity * Time.deltaTime;
+        controller.Move(_movement * speed * Time.deltaTime);
+        controller.Move(_velocity * Time.deltaTime);
+
+    }
+
+    /*Movement is partitioned into two methods for better input feedback.
+     Although in theory, the newq input master has the ability to handle 2D
```
Comment (dameos) [inline Project Dinner/Assets/PlayerMain.cs]: `newq` -> `new`

File: Project Dinner/Assets/PlayerMain.cs [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+        /*Creates sphere that checks if the player is touching the ground.*/
+        _isGrounded = Physics.CheckSphere(groundCheck.position, grounDistance, goundMask);
+
+        /* Condition to reset the Y velocity once the player is grounded.
+        When the player is grounded the velocity becomes negative if not reseted.*/
+        if (_isGrounded && _velocity.y < 0)
+        {   
+           /* -2f is used since it works a bit better than 0 to make sure the 
+            player is touching the ground */
+            _velocity.y = -2f;
```
Comment (dameos) [inline Project Dinner/Assets/PlayerMain.cs]: Make this a static constant, magic numbers are not ideal and hard to refactor

File: Project Dinner/Assets/PlayerMain.cs [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+    {
+        _moveAxis.x = ctx.ReadValue<float>();
+        //Debug.Log($"X axis {_moveAxis}");
+    }
+
+    void Jump(InputAction.CallbackContext ctx)
+    {
+        if (_isGrounded)
+        {
+            _velocity.y = Mathf.Sqrt(jumpHeight * -2f * gravity);
```
Comment (dameos) [inline Project Dinner/Assets/PlayerMain.cs]: Use here the constant I mentioned before

File: Project Dinner/Assets/PlayerCamera.cs [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+using UnityEngine.InputSystem;
+
+public class PlayerCamera : MonoBehaviour
+{
+    public Transform playerBody;
+    public float mouseSensitivity = 100f;
+
+    private InputMaster _controls;
+    private UnityEngine.Vector2 _direction;
+    private float xRotation = 0f;
```
Comment (dameos) [inline Project Dinner/Assets/PlayerCamera.cs]: Make this static

**Review by dameos — APPROVED** [review]
Some small remarks, other than that LGTM

File: Project Dinner/Assets/PlayerMain.cs (hunk 1 above)
Comment (pcanogo) [reply Project Dinner/Assets/PlayerMain.cs]: wtf. what a catch.

**Pull request merged**